### DISCUSSION TITLE
How do backups work?

### DISCUSSION THREAD

**Discussion by inteadit** [TandoorRecipes/recipes#257] [Q&A] [answered]
When I go to System there is a "Download Backup" button underneath "Backup & Restore". A few quick questions on this:

1.  How do I restore from a backup? I can't find the restore option anywhere. Must I do this manually by putting the json file in the right location in the server, or can it be done through the WebUI?
2.  The json file is small and only seems to contain the text ingredients/directions etc. for my recipes. Is there a way to backup the entire cookbook including the pictures used for each recipe?

I was not very smart and made the VM that hosts my recipes installation quite small. Before long it'll be full. I could always make a backup of the persistent storage volume from the VM and restore using that, I think - but an option to download a complete backup from the WebUI and restore from that would be nice as well.

Thank you!

**Comment by vabene1111** [comment, marked as answer]
To be honest, backups is a topic that i have not really had the time to create a good solution for. So while there is no convenient solution i can at least tell you how to manually backup everything that is required and how to best do so (or at least how i do it).

## The Database
Depending on your setup you will have postgresql, sqlite or any other database. The contents of the database are best stored by dumping it using the respective commands specific to the database system. I use postgresql with [this](https://github.com/vabene1111/DockerPostgresBackups) little tool to create regular backups and push them onto my nextcloud. I have to warn you tough, the project is not really polished and mostly for my personal use. You can also restore the json file generated by the website but you will have to put it into the webroot and run the `manage.py loaddata` as the backup is nothing else than a django fixture.

You could also just take the database files (e.g. tar them and copy/paste them) but i previously had some corruption problems using this approach so i cant recommend it.

## Images
All images are stored in the `mediafiles` directory. Simply copy it somewhere and restore it afterwards.

**Reply by V1d1o7** [reply]
Vabene, you mention in the docs pretty much the same thing here. Do you trust your script well enough to have a write up on how to use the backup script? If so i would be willing to write it up for you as i just went through the process of using it and it was fun trying to figure out part of it lol

**Reply by vabene1111** [reply]
That is a very good question. I have run the script for more than a year now, both in private and business and have also done some restores but i dont really feel like one should use it since it lacks tests and verification. 

If you want feel free to submit a PR over on the repo to update the readme so that it is easier to use but i dont really feel like putting it into the docs here, at least not until i have put some more effort into it

**Reply by V1d1o7** [reply]
I thought about updating the readme over there, but then wasn't sure if it was such a great idea since it isn't specific to this application.

**Reply by vabene1111** [reply]
i guess you could use this application as an example, it should be straight forward to update it for any other apllication

**Comment by bonelifer** [comment]
You could enable WIKI for this project and add it there with a disclaimer.
